deeplake/auto/unstructured/yolo/yolo.py

```diff
@@ -0,0 +1,329 @@
+
+        img_params["sample_compression"] = self.image_params.get(
+            "sample_compression", self.data.most_frequent_image_extension
+        )
+        name = self.image_params.get("name")
+
+        structure.add_first_level_tensor(
+            TensorStructure(
+                name=name,
+                params={i: img_params[i] for i in img_params if i != "name"},
+            )
+        )
+
+    def _ingest_data(self, ds: Dataset, progressbar: bool = True, num_workers: int = 0):
+        """Functions appends the the data to the dataset object using deeplake.compute"""
+
+        if self.image_creds_key is not None:
+            ds.add_creds_key(self.image_creds_key, managed=True)
+
+        # Wrap tensor data needed by the deeplake.compute function into a net dict.
+        tensor_meta = {
+            "images": ds[self.image_params["name"]].meta,
+            "labels": ds[self.label_params["name"]].meta,
+            "coordinates": ds[self.coordinates_params["name"]].meta,
+        }
+
+        @deeplake.compute
+        def append_data_bbox(data, sample_out, tensor_meta: Dict = tensor_meta):
+
+            # If the ingestion data is None, create empty annotations corresponding to the file
+            if data[1]:
+                yolo_labels, yolo_coordinates = self.data.read_yolo_coordinates(
+                    data[1], is_box=True
+                )
+            else:
+                yolo_labels = np.zeros((0))
+                yolo_coordinates = np.zeros((4, 0))
+
+            sample_out.append(
+                {
+                    self.image_params["name"]: self.data.get_image(
+                        data[0],
+                        tensor_meta["images"].is_link,
+                        self.image_creds_key,
+                    ),
+                    self.label_params["name"]: yolo_labels.astype(
+                        tensor_meta["labels"].dtype
+                    ),
+                    self.coordinates_params["name"]: yolo_coordinates.astype(
+                        tensor_meta["coordinates"].dtype
+                    ),
+                }
+            )
+
+        @deeplake.compute
+        def append_data_polygon(data, sample_out, tensor_meta: Dict = tensor_meta):
+
+            # If the ingestion data is None, create empty annotations corresponding to the file
+            if data[1]:
+                yolo_labels, yolo_coordinates = self.data.read_yolo_coordinates(
+                    data[1], is_box=False
+                )
+            else:
+                yolo_labels = np.zeros((0))
+                yolo_coordinates = []
+
+            sample_out.append(
+                {
+                    self.image_params["name"]: self.data.get_image(
+                        data[0],
+                        tensor_meta["images"].is_link,
+                        self.image_creds_key,
+                    ),
+                    self.label_params["name"]: yolo_labels.astype(
```

Comment:
In this case, the `astype` conversion results in coping of the array, although its small.
Can be solved by passing the type from the tensor upfront both to `np.zeros(4, dtype=...)` and the `read_yolo_coordinates`.

Reply:
Hey @ProgerDav Not sure I 100% get it. I want to cast at the end using `astype`, because I don't want to pass the `dtype` to `read_yolo_coordinates`. I want to assume that I don't know the `dtype` returned by `read_yolo_coordinates`, and then cast at the end. I see how I could populate the zeros with the correct `dtype` in advance, but I don't think I can do that with the output of `read_yolo_coordinates`, no?